### PR TITLE
Add property to catalog member to enabled hiding of legends

### PR DESCRIPTION
**Pull request by rowanwins**
Adds support for hiding legends on a catalog member by adding a `hideLegendInWorkbench` property.

For example
````
{
      "type": "wms",
      "name": "Cadastre History - Timeseries",
      "url": "https://portal.spatial.nsw.gov.au/server/services/Cadastre_History/MapServer/WMSServer",
      "hideLegendInWorkbench": true
}
````

### CODE REVIEW

**Comment by rowanwins** [comment]
I've pushed up a test file as per @soyarsauce 's pattern in #4133 however I renamed the extension from `.tsx` to `.tsxp` because TS wouldn't compile. 

The compile error in the test is
````
No overload matches this call.
  Overload 1 of 2, '(props: { forPrint: boolean; }, context?: any): ClassicComponent<{ forPrint: boolean; }, any>', gave the following error.
    Type '{ item: WebMapServiceCatalogItem; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<ClassicComponent<{ forPrint: boolean; }, any>> & Readonly<{ forPrint: boolean; }> & Readonly<...>'.
      Property 'item' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<ClassicComponent<{ forPrint: boolean; }, any>> & Readonly<{ forPrint: boolean; }> & Readonly<...>'.
  Overload 2 of 2, '(props: { forPrint: boolean; }, context?: any): Component<{ forPrint: boolean; }, any, any>', gave the following error.
    Type '{ item: WebMapServiceCatalogItem; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<{ forPrint: boolean; }, any, any>> & Readonly<{ forPrint: boolean; }> & Readonly<...>'.
      Property 'item' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<{ forPrint: boolean; }, any, any>> & Readonly<{ forPrint: boolean; }> & Readonly<...>'.

(JSX attribute) item: WebMapServiceCatalogItem
````
errors on this line
````
        const legendSection = <Legend item={wmsItem} />;
````

**Comment by na9da** [comment]
Looks great with tests in `tsx`!

File: test/ReactViews/Workbench/Controls/LegendSpec.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  let terria: Terria;
+  let wmsItem: WebMapServiceCatalogItem;
+
+  beforeEach(function() {
+    terria = new Terria({
+      baseUrl: "./"
+    });
+
+    wmsItem = new WebMapServiceCatalogItem("mywms", terria);
+    wmsItem.setTrait(
```
Comment (na9da) [inline test/ReactViews/Workbench/Controls/LegendSpec.tsx:19]: Wrap this in action?

Also the ones below.

Comment (soyarsauce) [reply test/ReactViews/Workbench/Controls/LegendSpec.tsx:19]: How come? setTrait is already an action itself, right?

Comment (na9da) [reply test/ReactViews/Workbench/Controls/LegendSpec.tsx:19]: Ah you're right! 
That was a more recent change I didn't realize. Thanks!

**Review by na9da — APPROVED** [review]

**Pull request merged**